packages/@aws-cdk/aws-appsync/lib/graphqlapi.ts

```diff
@@ -150,6 +160,38 @@ export interface OpenIdConnectConfig {
   readonly oidcProvider: string;
 }
 
+/**
+ * Configuration for Lambda authorization in AppSync. Note that you can only have a single AWS Lambda function configured to authorize your API.
+ */
+export interface LambdaAuthorizerConfig {
+  /**
+   * The authorizer lambda function.
+   * Note: This Lambda function must have the following resource-based policy assigned to it.
+   * When configuring Lambda authorizers in the console, this is done for you.
+   * To do so with the AWS CLI, run the following:
+   *
+   * `aws lambda add-permission --function-name "arn:aws:lambda:us-east-2:111122223333:function:my-function" --statement-id "appsync" --principal appsync.amazonaws.com --action lambda:InvokeFunction`
+   *
+   * @see https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-appsync-graphqlapi-lambdaauthorizerconfig.html
+   */
+  readonly handler: IFunction;
+
+  /**
+   * How long the results are cached.
+   * Disable caching by setting this to 0.
+   *
+   * @default Duration.minutes(5)
+   */
+  readonly resultsCacheTtl?: Duration;
+
+  /**
+   * A regular expression for validation of tokens before the Lambda function is called.
+   *
+   * @default - no regex filter will be applied.
+   */
+  readonly validationRegex?: string;
```

Comment:
can we also accept `RegEx` here? (and convert to `.source` underneath, checking what to do with flags)